pkg/ipamd/ipamd.go

```diff
@@ -187,9 +191,9 @@ var (
 type IPAMContext struct {
 	awsClient            awsutils.APIs
 	dataStore            *datastore.DataStore
-	k8sClient            kubernetes.Interface
+	standalonek8sClient  client.Client
+	k8sClient            client.Client
```

Comment:
As mentioned above - it smells like a bug if ipamd really requires both a cached and uncached client.  Can we consolidate these, or make the caching behaviour clearer?

Reply:
So, there are 3 different resources ipamd is acting on - `ENIConfigs`, `Nodes` (during ENI create flow) and `Pods` (during SGPP flow - to get pod annotations). We wanted to avoid a cached client for `Pods`(essentially avoid a watch on pod resource from every node) and so we're using a Cached client exclusively for `ENIConfigs` and `Nodes` and depend on raw client to get `Pod` objects.

Reply:
Yeah, we did discuss about it but opted to use cached client for the resources we depend upon during the ENI creation flow. No additional latency during ENI Create and not having to worry about retries if (for whatever reason) the API Server call fails at the most inopportune moment :). Also, the existing custom networking code is watching on Node label/annotation changes real time and adjusting the ENIConfig name applicable to that node although I am not sure if we actually recommend this. I think, we can revisit if this becomes inefficient as we start supporting clusters with 10k+ nodes.